src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -334,14 +334,13 @@ private static BoundCall CreateParameterlessCall(CSharpSyntaxNode syntax, BoundE
         internal sealed class AsyncForwardEntryPoint : SynthesizedEntryPointSymbol
         {
             // The user-defined asyncrhonous main method.
-            private MethodSymbol _userMain;
+            private readonly MethodSymbol _userMain;
             private DiagnosticBag _diagnosticBag;
```

Comment:
I've moved this bag to the constructor and provide the userMain location as the location for the diagnostic to go.